src/Board.ts

```diff
@@ -1,14 +1,72 @@
+import { Graph, Node, Board, DirectionInBoard, Deriv } from "./Types";
 
-export type Board = Array<Array<number>>;
+const DIRECTIONS: DirectionInBoard = {
+  E: {
+    x: 1,
+    y: 0,
+  },
+
+  S: {
+    x: 0,
+    y: 1,
+  },
+  W: {
+    x: -1,
+    y: 0,
+  },
+  N: {
+    x: 0,
+    y: -1,
+  },
+};
 
 export function getInitialBoard(): Board {
-    return [
-        [1, 1, 0, 0, 0, 2, 2],
-        [1, 1, 0, 3, 0, 2, 2],
-        [0, 0, 3, 3, 3, 0, 0],
-        [0, 3, 3, 3, 3, 3, 0],
-        [0, 0, 3, 3, 3, 0, 0],
-        [2, 2, 0, 3, 0, 1, 1],
-        [2, 2, 0, 0, 0, 1, 1]
-    ];
-}
\ No newline at end of file
+  return [
+    [1, 1, 0, 0, 0, 2, 2],
+    [1, 1, 0, 3, 0, 2, 2],
+    [0, 0, 3, 3, 3, 0, 0],
+    [0, 3, 3, 3, 3, 3, 0],
+    [0, 0, 3, 3, 3, 0, 0],
+    [2, 2, 0, 3, 0, 1, 1],
+    [2, 2, 0, 0, 0, 1, 1],
+  ];
+}
+
+export function canMoveMarbleInDirection(
+  boardGraph: Graph,
+  marblePosition: string,
+  direction: string
+): Boolean {
+  if (positionExistsInBoard(boardGraph, marblePosition)) {
```

Comment:
Maybe we can "reverse" the condition with || to have only one level of increment ?

Reply:
Or use && 
```
return positionExistsInBoard(boardGraph, marblePosition) && hasFreeSpotBeforeToMove(boardGraph, marblePosition, direction)
```

Reply:
has been discussed and resolved